cucumber-selenium-tests/src/test/java/com/skillstorm/PageObjects/TransactionPage.java

```diff
@@ -0,0 +1,281 @@
+
+
+    @FindBy(id = "sortByDropdown")
+    private WebElement sortByDropdown;
+
+
+    @FindBy(id = "directionDropdown")
+    private WebElement directionDropdown;
+
+
+    @FindBy(id = "addTransactionModal")
+    private WebElement addTransactionModal;
+
+
+    // Modal Fields
+    @FindBy(name = "vendorName")
+    private WebElement vendorNameField;
+
+    @FindBy(name = "accountId")
+    private WebElement accountDropdown;
+
+    @FindBy(name = "amount")
+    private WebElement amountField;
+
+    @FindBy(name = "category")
+    private WebElement categoryDropdown;
+
+    @FindBy(id = "addTransactionBtn")
+    private WebElement submitBtn;
+
+
+    // Filter Dropdowns
+    @FindBy(id = "allCategoriesDropDown")
+    private WebElement allCategoriesDropDown;
+
+
+    @FindBy(id = "allAccountDropDown")
+    private WebElement allAccountDropDown;
+
+
+    @FindBy(id = "allAmountsDropDown")
+    private WebElement allAmountsDropDown;
+
+
+    @FindBy(id = "allDatesDropDown")
+    private WebElement allDatesDropDown;
+
+
+    // Transaction Table Elements
+    @FindBy(id = "listOfTransactionsTitle")
+    private WebElement transactionsTableTitle;
+
+    @FindBy(xpath = "//*[@id=\"root\"]/div[1]/main/div/div[3]/div/table")
+    private WebElement transactionsTable;
+
+    @FindBy(xpath = "//*[@id=\"root\"]/div[1]/main/div/div[3]/div/table/tbody/tr[1]")
+    private WebElement transactionsTableFirstRow;
+
+    // Edit Transaction Elements
+    @FindBy(id = "editBtn")
+    private WebElement editBtn;
+
+    @FindBy(id = "editTransactionBtn")
+    private WebElement editSubmitBtn;
+
+    @FindBy(id = "edit-transaction-account")
+    private WebElement editTransactionAccountField;
+
+
+    @FindBy(id = "edit-transaction-category")
+    private WebElement editTransactionCategoryField;
+
+    // Delete Transaction Elements
+    @FindBy(id = "deleteBtn")
+    private WebElement deleteBtn;
+
+    private int tableSizeBeforeDeletion;
+
+    // Constructor
+    public TransactionPage(WebDriver driver) {
+        super(driver);
+        PageFactory.initElements(driver, this);
+    }
+
+    // Helper method to wait for element visibility
+    private WebElement waitForElement(WebElement element, int timeoutInSeconds) {
+        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(timeoutInSeconds));
+        return wait.until(ExpectedConditions.visibilityOf(element));
+    }
+
+    /**
+     * Helper method to pause execution for a specified amount of time.
+     * 
+     * @param milliseconds - the amount of time in milliseconds to pause
+     */
+    private void pause(long milliseconds) {
+        try {
+            Thread.sleep(milliseconds);
+        } catch (InterruptedException e) {
+            e.printStackTrace();
+        }
+    }
+
+
+    public void clickTab() {
+        DashboardNavbar navbar = new DashboardNavbar(driver);
+        navbar.clickButton(DashboardNavbar.BTN_TRANSACTIONS_NAME);
+        pause(2000);
```

Comment:
Low impact: The waitForElement method written in this file could have better loading times than pause(2000)